File: 백준/Silver/15663. N과 M （9）/N과 M （9）.swift
var nums = readLine()!.split(separator: " ").map {Int(String($0))!}
var numbers = readLine()!.split(separator: " ").map {Int(String($0))!}
numbers.sort()
var prior = [String]()
var selected = [String]()
var isVisited = Array(repeating: false, count: nums[0])

func DFS(start: Int) {
    if selected.count == nums[1] {
        print(selected.joined(separator: " "))
        return
    }
    var temp = ""
    for num in 0..<nums[0] {
        if isVisited[num] == false && temp != String(numbers[num]) {
            selected.append(String(numbers[num]))
            isVisited[num] = true
            temp = String(numbers[num])
            DFS(start: start+1)
            selected.removeLast()
            isVisited[num] = false
        }
    }
}

DFS(start:0)

File: 백준/Silver/15663. N과 M （9）/N과 M （9）.py
n, m = map(int, input().split())
nums = list(map(int, input().split()))
nums.sort()

lst = []
visited = [False] * n
numSet = set()

def DFS():
    if len(lst) == m:
        temp = " ".join(map(str, lst))
        if temp not in numSet:
            print(temp)
            numSet.add(temp)
        return
    for i in range(0, n):
        if not visited[i]:
            lst.append(nums[i])
            visited[i] = True
            DFS()
            lst.pop()
            visited[i] = False
DFS()